File: strands-kt/examples-java/src/main/java/Calculator.java
import com.strands.Agent;
import com.strands.ModelConfig;
import com.strands.StreamEvent;
import com.strands.Tool;
import java.util.List;

public class Calculator {
    public static void main(String[] args) {
        var calculator = new Tool(
            "calculator",
            "Evaluate a math expression and return the numeric result.",
            "{\"type\":\"object\",\"properties\":{\"expression\":{\"type\":\"string\",\"description\":\"A math expression to evaluate\"}},\"required\":[\"expression\"]}",
            (input, toolUseId) -> "{\"status\":\"success\",\"content\":[{\"text\":\"714\"}]}"
        );

        try (var agent = new Agent(
            ModelConfig.bedrock(),
            "You are a helpful assistant with a calculator tool.",
            List.of(calculator)
        )) {
            agent.forEachEvent("What is 42 * 17?", event -> {
                if (event instanceof StreamEvent.TextDelta td) {
                    System.out.print(td.getText());
                } else if (event instanceof StreamEvent.Stop stop) {
                    System.out.println("\n[" + stop.getReason() + "]");
                } else if (event instanceof StreamEvent.ToolUse tu) {
                    System.out.println("[tool-use: " + tu.getName() + "]");
                } else if (event instanceof StreamEvent.ToolResult tr) {
                    System.out.println("[tool-result: " + tr.getStatus() + " => " + tr.getContent() + "]");
                }
            }).join();
        }
    }
}

File: strands-kt/examples-kt/src/main/kotlin/Calculator.kt
import com.strands.Agent
import com.strands.ModelConfig
import com.strands.StreamEvent
import com.strands.Tool
import kotlinx.coroutines.runBlocking

fun main() = runBlocking {
    val calculator = Tool(
        name = "calculator",
        description = "Evaluate a math expression and return the numeric result.",
        inputSchema = """{"type":"object","properties":{"expression":{"type":"string","description":"A math expression to evaluate"}},"required":["expression"]}""",
    ) { _, _ ->
        """{"status":"success","content":[{"text":"714"}]}"""
    }

    Agent(
        model = ModelConfig.Bedrock(),
        systemPrompt = "You are a helpful assistant with a calculator tool.",
        tools = listOf(calculator),
    ).use { agent ->
        agent.stream("What is 42 * 17?").collect { event ->
            when (event) {
                is StreamEvent.TextDelta -> print(event.text)
                is StreamEvent.Stop -> println("\n[${event.reason}]")
                is StreamEvent.ToolUse -> println("[tool-use: ${event.name}]")
                is StreamEvent.ToolResult -> println("[tool-result: ${event.status} => ${event.content}]")
                else -> {}
            }
        }
    }
}
